From 82b2a5098ed3f079289ed5a0dce36ed0b9d71d8e Mon Sep 17 00:00:00 2001
From: Daniel Baluta <daniel.baluta@nxp.com>
Date: Mon, 12 Jun 2017 17:21:26 +0300
Subject: [PATCH 1884/5242] MLK-15067: ASoC: fsl: imx-wm8958: Don't fail if
 gpr is missing

commit  63fc4b11290c31712d7a1b6bc3ba55af66b857e5 from
https://source.codeaurora.org/external/imx/linux-imx.git

There are boards without gpr setting, so it's better not to fail
in such cases and only print a warning.

This is related to commit ce72b6d2668049 ("MLK-13574-1: ASoC: imx-wm8960:
remove the gpr dependency").

Signed-off-by: Daniel Baluta <daniel.baluta@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/imx-wm8958.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/sound/soc/fsl/imx-wm8958.c b/sound/soc/fsl/imx-wm8958.c
index c1d153d..df2e656 100644
--- a/sound/soc/fsl/imx-wm8958.c
+++ b/sound/soc/fsl/imx-wm8958.c
@@ -451,8 +451,7 @@ static int imx_wm8958_probe(struct platform_device *pdev)
 	ret = of_parse_phandle_with_fixed_args(pdev->dev.of_node, "gpr", 3,
 				0, &args);
 	if (ret) {
-		dev_err(&pdev->dev, "failed to get gpr property\n");
-		goto fail;
+		dev_warn(&pdev->dev, "failed to get gpr property\n");
 	} else {
 		data->gpr = syscon_node_to_regmap(args.np);
 		if (IS_ERR(data->gpr)) {
-- 
1.7.9.5

